tests/integration/compaction/src/test/java/org/apache/pulsar/tests/integration/TestCompaction.java

```diff
@@ -98,22 +88,16 @@ public void testPublishCompactAndConsumeCLI() throws Exception {
         }
     }
 
-    @Test
-    public void testPublishCompactAndConsumeRest() throws Exception {
-        PulsarClusterUtils.runOnAnyBroker(docker, clusterName,
-                                          PulsarClusterUtils.PULSAR_ADMIN, "tenants",
-                                          "create", "compaction-test-rest", "--allowed-clusters", clusterName,
-                                          "--admin-roles", "admin");
-        PulsarClusterUtils.runOnAnyBroker(docker, clusterName,
-                PulsarClusterUtils.PULSAR_ADMIN, "namespaces",
-                "create", "compaction-test-rest/ns1");
-        PulsarClusterUtils.runOnAnyBroker(docker, clusterName,
-                PulsarClusterUtils.PULSAR_ADMIN, "namespaces",
+    @Test(dataProvider = "ServiceUrls")
+    public void testPublishCompactAndConsumeRest(String serviceUrl) throws Exception {
+
+        this.createTenantName("compaction-test-rest", clusterName, "admin");
```

Comment:
add randomized name as suffix after `compaction-test-rest`

clusterName should pulsarCluster.getClusterName()